### PR TITLE
Fix incorrect commit creation

### PR DESCRIPTION
**Pull request by yashvardhannanavati**
Relates to FACTORY-2697

### CODE REVIEW

File: scrapers/koji.py [hunk 1]
```diff
@@ -218,9 +218,12 @@ def update_neo4j(self, builds):
             xml_root = ET.fromstring(task_dict['request'])
             commit_hash = None
             for child in xml_root.iter('string'):
+                # Avoids incorrect creation of Commits
+                if '.tar' in child.text:
+                    commit_hash = None
+                    break
                 if child.text and child.text.startswith('git'):
                     commit_hash = child.text.rsplit('#', 1)[1]
-                    break
```
Comment (mprahl) [inline scrapers/koji.py]: You want to keep this break statement.

File: scrapers/koji.py [hunk 2]
```diff
@@ -218,9 +218,12 @@ def update_neo4j(self, builds):
             xml_root = ET.fromstring(task_dict['request'])
             commit_hash = None
             for child in xml_root.iter('string'):
+                # Avoids incorrect creation of Commits
+                if '.tar' in child.text:
+                    commit_hash = None
+                    break
                 if child.text and child.text.startswith('git'):
```
Comment (mprahl) [inline scrapers/koji.py]: Change this to an else if for clarity.

File: scrapers/koji.py [hunk 3]
```diff
@@ -218,9 +218,12 @@ def update_neo4j(self, builds):
             xml_root = ET.fromstring(task_dict['request'])
             commit_hash = None
             for child in xml_root.iter('string'):
+                # Avoids incorrect creation of Commits
+                if '.tar' in child.text:
```
Comment (mprahl) [inline scrapers/koji.py]: This is to vague of a search. What if some metadata is stuffed in there that is unrelated and has ".tar" in it?

Comment (yashvardhannanavati) [reply scrapers/koji.py]: So I checked a couple of other tasks and they only had the commit hash so I decided to go ahead with this. I even looked through the data from teiid and there doesn't seem to be anything else which can distinguish a tar build. I am open to other ideas :)

File: scrapers/koji.py (hunk 1 above)
Comment (yashvardhannanavati) [reply scrapers/koji.py]: I removed the break statement because what if our other condition to check if there's a .tar is after we extract the hash (which isn't really hash)

Comment (mprahl) [reply scrapers/koji.py]: Oh I see... I think there's probably a better way to do this. Let me dig into it.

Comment (yashvardhannanavati) [reply scrapers/koji.py]: Another solution I can think of is after extracting the hash, check if it has special characters like '/' or '-'. Or if there's a specified length of the hash then we can check for that too. WDYT @mprahl ?

**Comment by yashvardhannanavati** [comment]
@mprahl We would not be able to use the approach you mentioned because the even the task with erroneous commit hash has this text `git://pkgs.devel.redhat.com/rpms/rhosp-director-images?#origin/rhos-10.0-rhel-7`

**Comment by mprahl** [comment]
@yashvardhannanavati but isn't that task not a "build" or "buildContainer" method? So my suggestion should work since it only looks at RPM builds and container builds.

**Comment by yashvardhannanavati** [comment]
@mprahl what are your thoughts on checking the `method` attribute instead which is present for all the tasks.

**Comment by mprahl** [comment]
@yashvardhannanavati, yes that is what I suggested in the diff a few comments above.

**Comment by yashvardhannanavati** [comment]
In the diff, you're checking for the `methodName` in the xml. I am suggesting the `method` attribute which should be present in the task_dict

**Comment by mprahl** [comment]
@yashvardhannanavati sure, either way is fine.
